third_party/amd/python/triton_amd.cc

```diff
@@ -161,6 +161,18 @@ void init_triton_amd(py::module &&m) {
       module->eraseNamedMetadata(openclVersion);
   });
 
+  m.def("disable_print_inline", [](llvm::Module *module) {
+    std::set<llvm::StringRef> noinlineFunctions = {
+        "__ockl_fprintf_stdout_begin",  "__ockl_fprintf_stderr_begin",
+        "__ockl_fprintf_append_args",   "__ockl_fprintf_append_string_n",
+        "__ockl_printf_begin",          "__ockl_printf_append_args",
+        "__ockl_printf_append_string_n"};
+
+    for (llvm::Function &f : module->functions())
+      if (f.hasName() && noinlineFunctions.count(f.getName()) != 0)
```

Comment:
What about checking whether the function has `__ockl_fprintf` or `__ockl_printf` prefix? This way we don't need to hardcode a list in the above and can be more future proof.

Reply:
I would like to be more conservative here to not forbid inlining for some unrelated functions. In general we want to inline as much as possible for better optimizations and performance.

These particular functions are special, because if kernel prints something this is probably for debug purposes and we do not expect top performance anyway,